data/instruct_advanced_mysql.csv

```diff
@@ -19,12 +19,12 @@ broker,mysql,instructions_date_join,"SELECT DATE_FORMAT(sbCustJoinDate, '%Y-%m')
+broker,mysql,instructions_date_join,"WITH active_customers AS (SELECT c.sbCustId, COUNT(t.sbTxId) AS num_transactions FROM broker.sbCustomer AS c JOIN broker.sbTransaction AS t ON c.sbCustId = t.sbTxCustId AND DATE_FORMAT(c.sbCustJoinDate, '%Y-%m') = DATE_FORMAT(t.sbTxDateTime, '%Y-%m') GROUP BY c.sbCustId) SELECT ac.sbCustId, c.sbCustName, ac.num_transactions FROM active_customers AS ac JOIN broker.sbCustomer AS c ON ac.sbCustId = c.sbCustId ORDER BY ac.num_transactions DESC LIMIT 1;","Which customer made the highest number of transactions in the same month as they signed up? Return the customer's id, name and number of transactions.","If transactions from 2 different tables from the same interval are to be joined, join on the respective truncated date fields eg `FROM t1 JOIN t2 ON DATE_FORMAT(t1.date, '%Y-%m-%d')=DATE_FORMAT(t2.date, '%Y-%m-%d').","To analyze the performance of stocks, join the tables for daily prices and tickers, filter for a designated time frame, and calculate the change in price To get the total transaction amount by each customer, you should join the customer and transaction tables, group by the customer, and sum up the amounts of the transactions If transactions from 2 different tables from the same interval are to be joined, join on the respective truncated date fields eg `FROM t1 JOIN t2 ON DATE_FORMAT(t1.date, '%Y-%m-%d')=DATE_FORMAT(t2.date, '%Y-%m-%d'). To identify the most popular stocks within a recent period, join the transaction table with the ticker table, filter for buy transactions during the last several days, group by the ticker symbol, and count the number of transactions."
```

Comment:
since the original instruction was to do a truncation of the date to just the month, perhaps you want to match the instructions to `DATE_FORMAT(t1.date, '%Y-%m')=DATE_FORMAT(t2.date, '%Y-%m')` without `%d`.

Reply:
Ok nice workaround